lib/awful/hotkeys_popup/widget.lua

```diff
@@ -274,7 +274,9 @@ function widget.new(args)
                 end
                 table.sort(
                     sorted_table,
-                    function(a,b) return (a.mod or '')..a.key<(b.mod or '')..b.key end
+                    function(a,b)
+                        local k1, k2 = a.key or a.keys[1][1], b.key or b.keys[1][1]
```

Comment:
shouldn't it be `join(a.keys[1])` instead of `a.keys[1][1]`?

Reply:
Currently, it looks like that:

![Screenshot_20191204_014332](https://user-images.githubusercontent.com/340384/70118865-0d472480-1637-11ea-840e-b6d6af0305de.png)

Is it correct?

As for the question itself, I guess the answer is that once the whole keyboard API matures, this code will need a serious cleanup since half of what it does (the whole reverse engineering part) will no longer be needed. Also the private APIs it uses will probably be fully obsolete and removed (like `awful.key.hotkeys`)

Reply:
as i understood, the sorting with [1][1] would be an issue only in case when hotkey have multiple `keys`